scripts/avalanche_script.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * PRIVATE_KEY={private_key} ts-node scripts/avalanche_script.ts
+ */
+import { ethers, providers, Contract } from 'ethers'
+import { waitForTx } from './utils/waitForTx'
+import { deployContract } from './utils/deployContract'
+
+import {
+  AvalancheTokenController__factory,
+  OwnedUpgradeabilityProxy__factory,
+  TrueUsd__factory,
+} from '../build'
+
+const txnArgs = { gasPrice: 100_000_000_000, gasLimit: 1_000_000 }
+const contractArgs = { gasPrice: 100_000_000_000, gasLimit: 6_000_000 }
+
+// this script was  originally used to transfer ownership of all avalanche TUSD to 0xf6E2Da7D82ee49f76CE652bc0BeB546Cbe0Ea521
+// this script can verify the owner & pending owners of the TUSD controller & token
+async function avalanche () {
+  const network = {
+    chainId: 43114,
+    name: 'ava-c-chain',
+  }
+
+  const provider = new providers.JsonRpcProvider('https://api.avax.network/ext/bc/C/rpc', network)
+  const wallet = new ethers.Wallet(process.env.PRIVATE_KEY, provider)
+
+  const controller = AvalancheTokenController__factory.connect('0xc3a247acE92A6A36FB69F70873A85fB8a66aDC1c', wallet)
+  const proxy = OwnedUpgradeabilityProxy__factory.connect('0xc3a247acE92A6A36FB69F70873A85fB8a66aDC1c', wallet)
+  const token = TrueUsd__factory.connect('0x1C20E891Bab6b1727d14Da358FAe2984Ed9B59EB', wallet)
+  const tokenProxy = OwnedUpgradeabilityProxy__factory.connect('0x1C20E891Bab6b1727d14Da358FAe2984Ed9B59EB', wallet)
+
+  // deploy, upgrade, set ratifiers
+  controller = await deployController(wallet)
+  await upgradeController(proxy, controller.address)
+  await setIsMintRatifier('0x083B59F244f8BAcbc79282Cdd623686324C962AC', 1, controller)
+  await setIsMintRatifier('0xb47DeA8731Fd846065294771Ab68B3Ee1FC6E880', 1, controller)
```

Comment:
I see on Etherscan that these addresses are used to call `TUSD.ratifyMint()` on mainnet. Do we have independent attestation that these are correct? I couldn't find them in the usual places where we track addresses internally.

Reply:
I can get additional confirmation from these addresses by having them send me AVAX

Reply:
confirmed. resolving this thread